stdlib/3/typing.pyi

```diff
@@ -404,6 +404,41 @@ class BinaryIO(IO[bytes]):
     @abstractmethod
     def __enter__(self) -> BinaryIO: ...
 
+class _ConcreteBinaryIO(BinaryIO):
```

Comment:
I don't think we should do this. Instead we should make some changes in io.pyi and inherit from io.BufferedRandom when we need a concrete subclass of BinaryIO.